### PR TITLE
Touchup Canary Platform

### PR DESCRIPTION
**Pull request by jamierocks**
- Don't refer to hooks as 'event's in parameter names
- Add icon to Neptune.inf files (optional plugin descriptor in NeptuneMod)
- Reinstate the MCP text format colouring (although for both canary + neptune this time)

### CODE REVIEW

File: src/main/kotlin/com/demonwav/mcdev/platform/canary/CanaryModule.kt [hunk 1]
```diff
@@ -50,6 +52,15 @@ class CanaryModule<out T : AbstractModuleType<*>>(facet: MinecraftFacet, overrid
         return canaryInf
     }
 
+    fun getNeptuneInf(): VirtualFile? {
```
Comment (stephan-gh) [inline src/main/kotlin/com/demonwav/mcdev/platform/canary/CanaryModule.kt]: You can implement this as regular property (using a custom getter and a private setter)

Same for the `canaryInf` above

**Review by stephan-gh — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/demonwav/mcdev/platform/canary/CanaryModule.kt [hunk 2]
```diff
@@ -31,23 +31,34 @@ import com.intellij.psi.search.GlobalSearchScope
 class CanaryModule<out T : AbstractModuleType<*>>(facet: MinecraftFacet, override val moduleType: T) : AbstractModule(facet) {
 
     override val type: PlatformType = moduleType.platformType
-    private var canaryInf: VirtualFile? = null
+    var canaryInf: VirtualFile? = null
+        get() {
+            if (field == null) {
```
Comment (stephan-gh) [inline src/main/kotlin/com/demonwav/mcdev/platform/canary/CanaryModule.kt]: Another thought: Do the null checks for `canaryInf` and `neptuneInf` in `setup()` (and initialize them if needed) and then call `setup()` every time. That way we can reduce duplication...

**Review by stephan-gh — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/demonwav/mcdev/platform/canary/CanaryModule.kt (hunk 2 above)
Comment (jamierocks) [reply src/main/kotlin/com/demonwav/mcdev/platform/canary/CanaryModule.kt]: huh, thought I'd already done that :S

**Review by stephan-gh — APPROVED** [review]
🎉

**Review by kashike — APPROVED** [review]

**Comment by DenWav** [comment]
I have an idea that I think will cleanup your lazy values, but I can see it being useful for every platform, so I will implement it after I merge.

**Pull request merged**